src/main/java/seedu/address/logic/parser/AddTagsCommandParser.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.AddTagsCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddTagsCommand object
+ */
+public class AddTagsCommandParser implements Parser<AddTagsCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddTagsCommand
+     * and returns an AddTagsCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public AddTagsCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        if (args.isEmpty() || args.length() < 4) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTagsCommand.MESSAGE_USAGE));
+        }
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(args.substring(0, 3));
+        } catch (IllegalValueException e) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTagsCommand.MESSAGE_USAGE));
+        }
+
+        List<String> tagsList = Arrays.asList(args.substring(3).split(" "));
```

Comment:
Magic numbers like `3` should be avoided and replaced with constants

Reply:
Yep, exactly what I meant previously haha. Maybe `ARGUMENTS_INDEX` or something?

Reply:
Will change to constant!